### ISSUE TITLE
Support IPv6 nameservers with DHCPv6

### ISSUE THREAD

**Issue by stealthybox** [weaveworks/ignite#440]
Currently, we are using this DHCPv4 implementation:
https://godoc.org/github.com/krolaw/dhcp4
This has the limitation of only supporting IPv4 DNS servers.

This looks like it might be a good candidate for DHCPv6:
https://godoc.org/github.com/insomniacslk/dhcp/dhcpv6

The modifier functions look easier to work with.
`WithDomainSearchList()` takes `[]string` instead of a single slice of `[]byte`. 
`WithDNS()` actually works with `[]net.IP`.

This would replace the DHCP server that runs inside of `ignite-spawn`.
See `pkg/container/dhcp.go`

**Comment by imazik** [comment]
@stealthybox  @chanwit I would like to give this a try.

**Comment by chanwit** [comment]
Surely, yes. Thank you @imazik !

**Comment by gpauloski** [comment]
@imazik @chanwit Is this still being worked on? I would like to work on it as well.

**Comment by stealthybox** [comment]
@gpauloski no it's not received any updates.
Please work on this if you still have the desire and bandwidth :)

**Comment by Nit123** [comment]
Hello, I am a student from the University of Texas at Austin. I wanted to know if a group of my colleagues and I could work on this issue as part of an assignment for my virtualization class as well as a recap on any work that's been done on the issue so far. Thanks in advance!

**Comment by gpauloski** [comment]
My group focused on some other open issues so we did not end up doing any work on this.

**Comment by supra08** [comment]
Is anyone working on this? I would like to take a stab at this.

**Comment by stealthybox** [comment]
@supra08, nobody is working on this to our knowledge.
Feel free to join our [dev call](https://docs.google.com/document/d/1fv8_WD6qXfvlIq7Bb5raCGyBvc42dNF-l8uaoZzoUYI/edit) on Mondays if you want help getting started!

**Comment by supra08** [comment]
Sure, that would be great!

**Comment by casibbald** [comment]
Do we have the same support for IPv4?

One would expect we should have this in place first, and is in fact valuable for another piece of work we have.


Charles Sibbald
Customer Success Engineer

charles.sibbald@weave.works
+44 7903 781 536

**Comment by stealthybox** [comment]
@casibbald yep, the ignite-spawn sandbox uses a temporary, in-process DHCPv4 server to do a one-time configuration of the firecracker linux guest for any needed IPv4 addresses.
